src/go/rpk/pkg/adminapi/api_partition.go

```diff
@@ -58,6 +60,20 @@ type ReconfigurationsResponse struct {
 	ReconciliationStatuses []Status  `json:"reconciliation_statuses"`
 }
 
+type ReplicaAssignment struct {
+	NodeID int `json:"node_id" yaml:"node_id"`
+	Core   int `json:"core" yaml:"core"`
+}
+
+type ClusterPartition struct {
+	Ns          string              `json:"ns" yaml:"ns"`
+	Topic       string              `json:"topic" yaml:"topic"`
+	PartitionID int                 `json:"partition_id" yaml:"partition_id"`
+	LeaderID    *int                `json:"leader_id,omitempty" yaml:"leader_id,omitempty"`
```

Comment:
Is the only field listed in the Redpanda Docs that does not come always in the response, if it's not a pointer and instead we use `int`, then it will show 0 when is not there. And 0 is a valid leader_id and it can confuse users.

Reply:
It might be worth leaving a comment to that effect to prevent future changes. I know if I were just looking at the current setup I'd have a nigh uncontrollable urge to make it not a pointer 😂

Reply:
+1 to comment, `// LeaderID may be missing in the response.`

Reply:
Comment added, thanks.